cmake/kokkos_compiler_id.cmake

```diff
@@ -94,6 +94,9 @@ if(Kokkos_ENABLE_HIP)
 
   string(REGEX MATCH "[0-9]+\\.[0-9]+\\.[0-9]+" TEMP_CXX_COMPILER_VERSION ${INTERNAL_COMPILER_VERSION_ONE_LINE})
   set(KOKKOS_CXX_COMPILER_VERSION ${TEMP_CXX_COMPILER_VERSION})
+  # get ROCm version
+  string(REGEX MATCH "roc-[0-9]+\\.[0-9]+\\.[0-9]+" INTERNAL_ROCM_VERSION_TMP ${INTERNAL_COMPILER_VERSION_ONE_LINE})
+  string(SUBSTRING ${INTERNAL_ROCM_VERSION_TMP} 4 -1 INTERNAL_ROCM_VERSION)
```

Comment:
can't we just use the hipcc or amdclang version? The rocm version should come from the findTPLRocm but that would need a pretty large update